src/net_processing.cpp

```diff
@@ -327,6 +327,11 @@ struct Peer {
         LOCK(m_tx_relay_mutex);
         return m_can_tx_relay ? m_tx_relay.get() : nullptr;
     };
+    const TxRelay* GetTxRelay() const LOCKS_EXCLUDED(m_tx_relay_mutex)
+    {
+        LOCK(m_tx_relay_mutex);
+        return m_can_tx_relay ? m_tx_relay.get() : nullptr;
+    };
```

Comment:
we could do one `TxRelay* GetTxRelay() const LOCKS_EXCLUDED(m_tx_relay_mutex)` instead of having two of these

Reply:
how?

Reply:
if the function is const, we can only return a const pointer to a member no?

Reply:
hmm I don't think there is a rule like that... const function simply ensures you don't modify members _inside of it_, what you do with a returned (pointer to a) member is up to you.

Reply:
But does that make sense? It does indeed compile, but if I have a 

```
const Peer& p;
```

does it make sense that I can do 

```
p.GetTxRelay()->m_relay_txs = false?
```